smack-experimental/src/main/java/org/jivesoftware/smackx/httpfileupload/HttpFileUploadManager.java

```diff
@@ -494,11 +495,12 @@ private void upload(InputStream iStream, long fileSize, Slot slot, UploadProgres
             case HttpURLConnection.HTTP_NO_CONTENT:
                 break;
             default:
-                throw new IOException("Error response " + status + " from server during file upload: "
-                                + urlConnection.getResponseMessage() + ", file size: " + fileSize + ", put URL: "
-                                + putUrl);
+                throw new HttpUploadException(status, urlConnection.getResponseMessage(), fileSize, putUrl);
```

Comment:
This should be an extra subclass of `HttpUploadException`, which itself, should become an abstract class.

Reply:
Made the class abstract, and made two concrete and more specific exceptions